pkg/operator/csi/csidrivercontrollerservicecontroller/csi_driver_controller_service_controller.go

```diff
@@ -217,6 +231,50 @@ func (c *CSIDriverControllerServiceController) sync(ctx context.Context, syncCon
 	return err
 }
 
+func (c *CSIDriverControllerServiceController) syncDeleting(opSpec *opv1.OperatorSpec, opStatus *opv1.OperatorStatus, ctx context.Context, syncContext factory.SyncContext) error {
+	klog.V(4).Infof("syncDeleting")
+	deployment, err := c.getDeployment(opSpec)
+	if err != nil {
+		return err
+	}
+
+	err = c.kubeClient.AppsV1().Deployments(deployment.Namespace).Delete(ctx, deployment.Namespace, metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	} else {
+		klog.V(2).Infof("Deleted Deployment %s/%s", deployment.Namespace, deployment.Name)
+	}
+
+	// All removed, remove the finalizer as the last step
+	err = c.operatorClient.RemoveFinalizer(c.name)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *CSIDriverControllerServiceController) getDeployment(opSpec *opv1.OperatorSpec) (*appsv1.Deployment, error) {
```

Comment:
@ravisantoshgudimetla you had a PR trying to collapse this functionality for deployments.  Is it ready to merge?  You're racing this PR now.

Reply:
It has been merged